This PR completes the Milestone Migrate Project Explorer

timbittner opened the pull request:
All functions of the original SWT ProjectExplorer are now available in the FX ProjectExplorer.

New features since last PR:

- DoubleClickListener to open folder nodes and special files (file endings defined by other plugins) https://github.com/Jooshua/e4CompareFramework-team2/issues/11
- DropTarget to quickly import files from the OS Explorer into the project (works with multiple files and empty folders) https://github.com/Jooshua/e4CompareFramework-team2/issues/10
- Workspace Structure Setup (Extensionpoints available to other projects, setting up the initial tree structure) https://github.com/Jooshua/e4CompareFramework-team2/issues/13
- Refactor Codebase to conform to FX Model-View-Controller pattern https://github.com/Jooshua/e4CompareFramework-team2/issues/19

Note that the last bullet point only moves existing logic between files, while refactoring Class and Package names.

kyshidar committed:
Migrate Double Click Listener
Task #11

kyshidar committed:
Migrate SWT DragListener to FX; Import is still missing the target directory from the event

Commit:
Add CustomTreeCell

10: Migrate ProjectExplorerDropTarget

Task-Url: https://github.com/Jooshua/e4CompareFramework/issues/10

valentinpurrucker committed:
Add initial setup of workspace structure.

Task: #13

Commit:
Merge branch 'feature/task_10-Migrate_DropTarget' into team02

Conflicts:
	de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/ProjectExplorerViewController.java

Commit:
Merge branch 'task_11-Migrate_Double_Click_OpenFile_Listener' into
team02

Conflicts:
	de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/ProjectExplorerView.java
	de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/ProjectExplorerViewController.java

timbittner committed:
Rename ViewController classes and move logic accordingly.

Task #19

TODO: Check instance holding of new ViewController

timbittner committed:
Move ViewController and Adapter to Controller Package.

Also improve method parameter passing.

Task #19

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/CustomTreeCell.java:
```diff
@@ -0,0 +1,63 @@
+	public CustomTreeCell(WorkspaceFileSystem workspaceFileSystem) {
+		super();
+
+		setOnDragOver(new EventHandler<DragEvent>() {
```
use lambda expression e-> {
}

kyshidar replied:
has been updated and tested

KamilRosiak reviewed (changes requested):
Benutzt den Autoformatter

paulo-haas commented on de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/listeners/OpenFileListener.java:
```diff
@@ -1,34 +1,56 @@
+	private void openFile(FileTreeElement target) {
+		if(target != null) {
+			String path = target.getAbsolutePath();
+			String fileExtension = path.substring(path.lastIndexOf(".")+1);
```
`getExtension()` is provided by FileTreeElement

kyshidar replied:
improvement has been applied

paulo-haas commented on de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/listeners/OpenFileListener.java:
```diff
@@ -1,34 +1,56 @@
+		if(event.getButton().equals(MouseButton.PRIMARY) && event.getClickCount() == 2) {
+			// should not trigger on directories
+			FileTreeElement target = services.rcpSelectionService.getCurrentSelectionFromExplorer();
+			if(!target.isDirectory()) {
```
Does this work when `target` equals `null`?

kyshidar replied:
now also checks target for null

kyshidar committed:
Changes due to review:
Change CustomTreeCell DragHandler to lambda expressions;
Apply proposed changes to OpenFileListener; Autoformat all
ProjectExplorer Files;

kyshidar commented:
the list of changes is long, but aside from the proposed changes everything else was done by the autoformatter and should not require further review

Blum-P commented on de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/ProjectExplorerLabelProvider.java:
```diff
@@ -52,34 +55,36 @@ public boolean isLabelProperty(Object element, String property) {
 					image = imageService.getImage(null, "icons/Explorer_View/items/file16.png");
-				};
+				}
+				;
```
Are you sure to let that semicolon there?

Jooshua replied:
Resolved :)

Commit:
Move semicolon in ProjectExplorerLabelProvider
